### ISSUE TITLE
Incoming emails - no support for explicit TLS or STARTTLS

### ISSUE THREAD

**Issue by waaeh** [waaeh/FakeOpenSmtpRelay#2]
Sending an email to the honeypot over tcp/465 or using tcp/25 and requesting STARTTLS does not work.

The root cause is probably in INetSim. INetSim claims to support both scenarios. [Some attempts to isolate and fix the issue have already been made, without success so far](/waaeh/FakeOpenSmtpRelay/tree/master/INetSim#work-in-progress---adaptations-to-support-starttls--explicit-smtps).

**Comment by waaeh** [comment]
As linked above, getting rid of the limitation to SSLv2 & SSLv3 doesn't solve the issue unfortunately:
File /usr/share/perl5/INetSim/SMTP.pm:
```diff
	sub upgrade_to_ssl {
		my $self = shift;
+		# EDITED to exclude the SSL version stuff
-		my %ssl_params = (  SSL_version             => "SSLv23",
-		                    SSL_cipher_list         => "ALL",
-		                    SSL_server              => 1,
-		                    SSL_use_cert            => 1,
-		                    SSL_key_file            => $self->{ssl_key},
-		                    SSL_cert_file           => $self->{ssl_crt} );
+		my %ssl_params = ( SSL_server              => 1,
+		                    SSL_use_cert            => 1,
+		                    SSL_key_file            => $self->{ssl_key},
+		                    SSL_cert_file           => $self->{ssl_crt} );
+		# ADDED for debugging purposes
+		$IO::Socket::SSL::DEBUG = 3;
```

**Comment by waaeh** [comment]
The absence of STARTTLS might be more problematic than initially thought. [As documented in this bug](https://github.com/aio-libs/aiosmtpd/pull/66#issuecomment-315148824), AUTH LOGIN should not be allowed over an unencrypted channel.

The following probe sequence was observed in the wild:
- Client connects over SMTP tcp/25 and sends `EHLO User`
- INetSim replies with the capability list, without STARTTLS 
- Client sends `RSET`
- INetSim acknowledges reset
- Client asks for `AUTH LOGIN`
- INetSim initiates the authentication, in violation of the RFC
- Client asks to `QUIT`

Either the client detects that the server is not RFC conform and thus a honeypot, or the client has some internal logic to respect the RFC. In both cases, we're missing out on probes / spams due to the missing STARTTLS feature.

**Comment by waaeh** [comment]
Resolved in version 1.0 and the usage of aiosmtpd instead of INetSim.
